app/models/king.rb

```diff
@@ -1,7 +1,68 @@
+# rubocop:disable Metrics/AbcSize
 class King < Piece
   def move_legal?(to_row, to_col)
     row_diff = (row - to_row).abs
     col_diff = (col - to_col).abs
     row_diff <= 1 && col_diff <= 1
+    # add castling moves to the 'whitelist', then verify the validity of the castle
+    #   further down in the logic
+  end
+
+  # override to enable castling
+  def finalize_move!(to_row, to_col)
+    return false if !get_attackers(to_row, to_col).empty?
+    move_to!(to_row, to_col)
+    # for castling something like: move_the_rook if castling?
+  end
+
+  def in_check?
+    !get_attackers(row, col).empty?
```

Comment:
If you want a single method for `!something.empty?`, ActiveSupport has added `.present?` ([definition](https://github.com/rails/rails/blob/bbb84a17213d60f25a61ae4a09111d5adac01a05/activesupport/lib/active_support/core_ext/object/blank.rb#L21-L23)).  You can only use it in a Rails project since it's added by ActiveSupport and not built into Ruby, but this is probably a good use-case for it.